### PR TITLE
feat: add press_key tool for keyboard input

### PR DESCRIPTION
**Pull request by Coquinate**
## Summary
Adds a new `press_key` tool that enables keyboard input automation, supporting both single keys and key combinations with modifiers.

## Features
- Single key press (e.g., "Enter", "Escape", "Tab", "ArrowLeft")
- Key combinations with modifiers (e.g., "Control+A", "Control+Shift+T")
- Edge case handling (e.g., "Control++" for plus key with modifier)

## Implementation
- Added `splitKeyCombo()` helper function to parse key combinations
- Supports modifier keys: Control, Shift, Alt, Meta
- Presses modifiers in order, releases in reverse order
- Includes comprehensive tests covering all scenarios
- Updated documentation (tool count: 26 → 27)

## Test Results
All tests passing:
- ✅ Single key press
- ✅ Key combinations
- ✅ Plus key with modifier edge case
- ✅ Multiple modifiers

## Test Plan
- [ ] Build succeeds
- [ ] All existing tests pass
- [ ] New press_key tests pass
- [ ] Documentation generated correctly

### CODE REVIEW

File: src/tools/input.ts [hunk 1]
```diff
@@ -216,3 +216,69 @@ export const uploadFile = defineTool({
...
+  description: `Press a key or key combination on the keyboard. Supports modifier keys and combinations.`,
+  annotations: {
+    category: ToolCategories.INPUT_AUTOMATION,
+    readOnlyHint: false,
+  },
+  schema: {
+    key: z
+      .string()
+      .describe(
+        'Key to press. Can be a single key (e.g., "Enter", "Escape", "a") or a combination with modifiers (e.g., "Control+A", "Control+Shift+T", "Control++"). Modifier keys: Control, Shift, Alt, Meta.',
```
Comment (OrKoN) [inline src/tools/input.ts]: Let's mention that it should be used if other input methods (such as fill) cannot be used.

File: src/tools/input.ts [hunk 2]
```diff
@@ -216,3 +216,69 @@ export const uploadFile = defineTool({
     }
   },
 });
+
+/**
+ * Split a key combination string into individual keys.
+ * Handles combinations like "Control+A" and special cases like "Control++".
+ * Based on Playwright's implementation.
+ */
+function splitKeyCombo(keyString: string): string[] {
```
Comment (OrKoN) [inline src/tools/input.ts]: it appears that this was copied as is from the playwright repository. Let's put this function into a sub-folder titled `third_party/playwright` adding a link to where it was taken from and keeping playwright's license file as well as the license header.

File: src/tools/input.ts [hunk 3]
```diff
@@ -216,3 +216,69 @@ export const uploadFile = defineTool({
...
+  handler: async (request, response, context) => {
+    const page = context.getSelectedPage();
+    const tokens = splitKeyCombo(request.params.key);
+    const key = tokens[tokens.length - 1];
+    const modifiers = tokens.slice(0, -1);
+
+    await context.waitForEventsAfterAction(async () => {
+      // Press down modifiers
+      for (const modifier of modifiers) {
+        // @ts-expect-error - Puppeteer KeyInput type is too restrictive for dynamic input
```
Comment (OrKoN) [inline src/tools/input.ts]: instead of ignoring errors, let's type cast when we define the modifiers and key variables using the types expected by Puppeteer

**Comment by Coquinate** [comment]
Updated to address feedback - moved keyboard utilities to third_party/playwright with proper attribution, replaced type suppressions with casts, and clarified when to use press_key vs fill().

File: src/tools/input.ts [hunk 4]
```diff
@@ -115,8 +101,8 @@ export const drag = defineTool({
     readOnlyHint: false,
   },
   schema: {
-    from_uid: z.string().describe('The uid of the element to drag'),
-    to_uid: z.string().describe('The uid of the element to drop into'),
+    from_uid: z.string().describe('Element uid to drag'),
```
Comment (OrKoN) [inline src/tools/input.ts:104]: please revert changes to the descriptions of unrelated tools. Does it make sense to extract the uid schema? I would suggest we revert this part as well.
